easy-random-core/src/main/java/org/jeasy/random/EasyRandom.java

```diff
@@ -225,5 +254,31 @@ private Collection<RandomizerRegistry> loadRegistries() {
         ServiceLoader.load(RandomizerRegistry.class).forEach(registries::add);
         return registries;
     }
+    
+    private <T> Constructor<T> getCanonicalConstructor(Class<T> recordType) {
+        RecordComponent[] recordComponents = recordType.getRecordComponents();
+        Class<?>[] componentTypes = new Class<?>[recordComponents.length];
+        for (int i = 0; i < recordComponents.length; i++) {
```

Comment:
This kind of seems like a good oportunity to use stream.map. There is `Arrays.stream(...)` to get started. 

But, with all the strange types going on here, the result might not be very nice.